app/models/content_item.rb

```diff
@@ -10,9 +10,23 @@ def self.create_or_update!(attributes, organisation)
     content_id = attributes.fetch(:content_id)
     content_item = self.find_or_create_by(content_id: content_id)
 
-    attributes = attributes.slice(*content_item.attributes.symbolize_keys.keys)
+    content_item.add_organisation(organisation)
+    content_item.add_taxonomies(attributes.fetch(:taxons))
 
+    attributes = attributes.slice(*content_item.attributes.symbolize_keys.keys)
```

Comment:
- Nice to have: `content_item.attributes.symbolize_keys.keys` this can have its own wrapper method and kimprove readability